src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -682,12 +682,27 @@ TransactionManager::get_extents_if_live(
             auto &pin_seg_paddr = pin_paddr.as_seg_paddr();
             auto pin_paddr_seg_id = pin_seg_paddr.get_segment_id();
             auto pin_len = pin->get_length();
-            if (pin_paddr_seg_id != paddr_seg_id) {
+            if (bool pin_valid = pin_seg_paddr >= paddr &&
+                  pin_seg_paddr.add_offset(pin_len) <= paddr.add_offset(len);
+                pin_paddr_seg_id != paddr_seg_id || !pin_valid) {
```

Comment:
This can be reverted to the original `if (pin_paddr_seg_id != paddr_seg_id) {`, right?

Reply:
yes, once https://tracker.ceph.com/issues/66941 is implemented.

Reply:
Is it better to only comment out the `ceph_assert` in this step to avoid the direct crash, and make the behavior mostly correct by reclaiming the unexpected extent?

Reply:
Um, sorry, I don't quite follow. Why allow rewriting an extent that's not within the range of the reclaiming when fixing it isn't very complex?

Reply:
> Why allow rewriting an extent that's not within the range of the reclaiming when fixing it isn't very complex?

Probably there is some misunderstanding, and yeah, I agree we should only rewrite the extent within the range of the reclaiming segment.

My suggestion was that:
```c++
if (pin_paddr_seg_id != paddr_seg_id) {
  return seastar::now;
}
// ceph_assert(pin_seg_paddr >= paddr &&
//             pin_seg_paddr.add_offset(pin_len) <= paddr.add_offset(len));
<reclaim all the extents within the range of the reclaiming segment (pin_paddr_seg_id == paddr_seg_id),
 including the extents not in the range [paddr~len], and after the next fix,
 uncomment the above ceph_assert.>
```

Not sure where I might be missing?

Reply:
Done:-)